### PR TITLE
PromQL Builder Explainer: Added promql documentation and updated prompt

### PR DESCRIPTION
**Pull request by edwardcqian**
**What is this feature?**

- Added promql function descriptions into promql explainer functionality
- Updated prompt structure
- Updated prompt to end with `This query is trying to answer the question:`, this will help the generated output to fit better with the UI

**Why do we need this feature?**

Current promql explainer functionality do not include function documentation which was already available in promql code

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [x] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Comment by edwardcqian** [comment]
Note: not to merge until post PromCon (Sep 28, 29)

File: public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts [hunk 1]
```diff
@@ -40,15 +39,51 @@ interface TemplateSearchResult {
...
+//   metricName,
+//   metricType,
+//   metricMetadata,
+//   query,
+// }: ExplainUserPromptParams): llms.openai.Message[] {
+//   return [
+//     { role: 'system', content: ExplainSystemPrompt },
+//     { role: 'user', content: GetExplainUserPrompt({ documentation, metricName, metricType, metricMetadata, query }) },
+//   ];
+// }
```
Comment (gerboland) [inline public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts:53]: Why did you keep the above?

File: public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts [hunk 2]
```diff
@@ -40,15 +39,51 @@ interface TemplateSearchResult {
...
+  datasource: PrometheusDatasource,
+): llms.openai.Message[] {
+
+  let metricMetadata = '';
+  let metricType = '';
+
+  const pvq = buildVisualQueryFromString(query);
+
+  if (datasource.languageProvider.metricsMetadata) {
+    metricType = getMetadataType(metric, datasource.languageProvider.metricsMetadata!) ?? '';
```
Comment (gerboland) [inline public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts:54]: `datasource.languageProvider.metricsMetadata` shouldn't need the `!` in this branch.

File: public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts [hunk 3]
```diff
@@ -40,15 +39,38 @@ interface TemplateSearchResult {
...
+    metricMetadata = getMetadataHelp(metric, datasource.languageProvider.metricsMetadata!) ?? '';
+  }
+
+  const documentationBody = pvq.query.operations.map((op) => {
+    const def = promQueryModeller.getOperationDef(op.id);
+    if (!def) {
+      return '';
+    }
+    const title = def.renderer(op, def, '<expr>');
+    const body = def.explainHandler ? def.explainHandler(op, def) : def.documentation ?? 'no docs';
```
Comment (gerboland) [inline public/app/plugins/datasource/prometheus/querybuilder/components/promQail/state/helpers.ts:64]: Would it be better to not mention docs at all in the prompt, if the docs are missing?

**Review by gerboland — COMMENTED** [review]
Couple of little things, but happy otherwise!

**Review by gerboland — APPROVED** [review]

**Pull request merged**